examples/federated_learning_with_encryption.py

```diff
@@ -182,34 +177,27 @@ def predict(self, X):
         return X.dot(self.weights)
 
     def encrypted_gradient(self, sum_to=None):
-        """Compute gradient. Encrypt it.
+        """Compute and encrypt gradient.
+
         When `sum_to` is given, sum the encrypted gradient to it, assumed
         to be another vector of the same size
         """
-
-        gradient = encrypt_vector(self.pubkey, self.compute_gradient())
+        encrypted_gradient = encrypt_vector(self.pubkey, self.compute_gradient())
 
         if sum_to is not None:
-            if len(sum_to) != len(gradient):
-                raise Exception('Encrypted vectors must have the same size')
-            return sum_encrypted_vectors(sum_to, gradient)
+            return sum_encrypted_vectors(sum_to, encrypted_gradient)
         else:
-            return gradient
+            return encrypted_gradient
 
 
-if __name__ == '__main__':
-
-    # Learning params
-    n_iter, eta = 50, 0.01
-
-    names = ['Hospital 1', 'Hospital 2', 'Hospital 3']
-    n_clients = len(names)
+def federated_learning(n_iter, eta, n_clients, key_length):
```

Comment:
I don't think it detracts readability - and it makes it easier to experiment with different number of clients and different key lengths.